.github-workflow-script

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env python3
+import argparse
+import yaml
+
+# Parse command-line arguments
+parser = argparse.ArgumentParser()
+parser.add_argument("workflow_file",
+                    help="Path to the GitHub Actions workflow file")
+
+args = parser.parse_args()
+
+# Load the GitHub Actions workflow file as YAML
+with open(args.workflow_file, "r") as file:
+    workflow = yaml.safe_load(file)
+
+# Get the jobs for "check_python" and "check_auxiliary_files"
+check_python = workflow["jobs"]["check_python"]
+check_aux_files = workflow["jobs"]["check_auxilary_files"]
+
+# Extract the shell commands from each job's steps
+check_python_steps = [step['run'] for step in workflow['jobs']
+                                     ['check_python']['steps']
+                     if 'run' in step and any(name in step['name']
+                        for name in ['Check python sources with flake8',
+                                     'Check python sources with pylint',
+                                     'Test with pytest'])
+                      ]
+
+check_aux_files_commands = [step['run'] for step in workflow['jobs']
+                                          ['check_auxiliary_files']['steps']
+                            if 'run' in step and any(name in step['name']
+                              for name in ['Check YAML files with yamllint'])
+                            ]
```

Comment:
This is a bit too tightly coupled to the workflow YAML we have, and so is brittle (will break easily with future changes).

How about we turn this around: we mark the steps that should be used in pre-commit hooks in the workflow YAML somehow, and then pick them out in this script. So you could just go over each job and step in order, pick the ones that are marked, and output their commands.

We could mark the steps by assigning a special environment variable for each of them. E.g.:
```yaml
      - name: Check python sources with flake8
        run: "flake8 kcidb *.py"
        env:
          KCIDB_HOOKS: "pre-commit"
```

Reply:
Another thing: I don't think we need to store the steps in arrays first. Just output them as you find them. It would also be nice to have the name of each step output as a comment in front of its command. So the example above would be output like this:
```bash
# Check python sources with flake8
flake8 kcidb *.py
```

Reply:
> So you could just go over each job and step in order, pick the ones that are marked, and output their commands.

That seems interesting. I will try it tonight.

Reply:
> ```shell
> # Check python sources with flake8
> ```

```
#Check python sources with flake8:
flake8 kcidb *.py

#Check python sources with pylint:
pylint kcidb *.py

#Test with pytest:
KCIDB_IO_HEAVY_ASSERTS=1 KCIDB_HEAVY_ASSERTS=1 pytest --tb=native

#Check YAML files with yamllint:
find -name '*.yaml' -or -name '*.yml' | xargs yamllint
```
that # names will be ignored right?
this is the terminal result

Reply:
The hash character `#` starts a comment in shell (same as in Python). Please put a space after it, to make the comments easier to read, and please don't add the colon `:` on the end. They're fine without them. Everyone expects a comment on a line by itself to describe the following line(s), and the colon just creates unnecessary visual noise.

Reply:
Otherwise looks good!

Reply:
```
# Check python sources with flake8
# Check python sources with pylint
# Test with pytest
================================================================= test session starts =================================================================
platform linux -- Python 3.10.6, pytest-7.2.2, pluggy-1.0.0
rootdir: /home/shivam/Desktop/pre-commit/kcidb
collected 72 items                                                                                                                                    

test_kcidb.py .....^C
```

is this good?  Can you check the PR I used the printed commands in the output of github-workflow-script and checked if the output starts with # and printed that out in the pre-commit file And executed the command below it